server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -250,6 +250,12 @@
     public static final Setting<Integer> MAX_SLICES_PER_SCROLL = Setting.intSetting("index.max_slices_per_scroll",
         1024, 1, Property.Dynamic, Property.IndexScope);
 
+    /**
+     * The maximum length of regex string allowed in a regex query.
+     */
+    public static final Setting<Integer> MAX_REGEX_LENGTH_SETTING = Setting.intSetting("index.max_regex_length",
+        1000, 1, Property.Dynamic, Property.IndexScope);
```

Comment:
We should also set the maximum value for this setting. `1000` should be default and possible max value for this setting. Otherwise we will get a Lucene error: https://github.com/apache/lucene-solr/blob/master/lucene/core/src/java/org/apache/lucene/util/automaton/Operations.java#L1034-L1037

I wonder if we should even make this setting updatable, just set it to `1000`.  Then we could detect the error earlier during the `RegexpQueryBuilder` constructor (before query execution), and error message to the user would be cleaner.

Reply:
The recursion limit is based on the longest string that is accepted by the automaton so a regex can potentially have more than `1000` chars without breaking. IMO it's ok to have a soft limit here and only test when we build the Lucene query.

Reply:
@jimczi  Thanks for the suggestions. @liketic  let's leave it as Jim suggested, and you already implemented.